### ISSUE TITLE
Run mutants multithreaded

### ISSUE THREAD

**Issue by hugo-vrijswijk** [stryker-mutator/stryker4s#10]
To improve performance, mutants can be ran multithreaded. Should be dependent on the number of cores on the machine or set via a configuration option

**Comment by legopiraat** [comment]
We could make it an optional configuration field. 

By default we could use something like `int cores = Runtime.getRuntime().availableProcessors() - 1`

the minus one is on purpose so we dont fully kill the machine's performance.

**Comment by hugo-vrijswijk** [comment]
I agree with both of those. Having cores - 1 also leaves room for an orchestrator of some kind.

We should probably also let the user know how many test runners are started, and based on what (config or processor count), same as Stryker

**Comment by hugo-vrijswijk** [comment]
It looks like SBT [runs all tests in parallel by default](https://www.scala-sbt.org/1.x/docs/Testing.html#Disable+Parallel+Execution+of+Tests). Not sure how Maven does this, but that could mean this might be a slightly lower priority for now. There might not be that big of a performance boost from running in parallel (or it might have a negative impact on performance with lots of timeouts)

**Comment by legopiraat** [comment]
Agreed for sure if we could leave the sbt process open it would be a huge gain. Am trying to come up with a way to remove sbt from the testruns in full. But it has some complications on the side which i need to investigate.

**Comment by cbrunnkvist** [comment]
There are several reasons to why a project can decide _not_ run local tests in parallel... 

A more scalable approach than just local parallelism would perhaps be some method of deciding a "partition ID", so that we can use CI pipelines to construct as much parallelism as we want. This would be fairly advanced to set up, but it would be quite useful to _have_ that option: The One Thing that stops this from being even remotely possible at the moment is _the lack of separation_ between _running the mutation tests_ and _generating the human-readable report_ (i.e. akin to how [sbt-scoverage](https://github.com/scoverage/sbt-scoverage) separates `sbt coverage` from `sbt coverageReport`).
